components/brave_wallet/common/f4_jumble.cc

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/common/f4_jumble.h"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+
+#include "base/notreached.h"
+#include "brave/third_party/argon2/src/src/blake2/blake2.h"
+
+namespace brave_wallet {
+
+namespace {
+
+using Blake2bPersonalBytes = std::array<uint8_t, BLAKE2B_PERSONALBYTES>;
+
+// Sizes for Blake2b are defined here https://zips.z.cash/zip-0316#solution
+constexpr size_t kMinMessageSize = 48u;
+constexpr size_t kMaxMessageSize = 4184368u;
+constexpr size_t kLeftSize = 64u;
+static_assert(kLeftSize <= BLAKE2B_OUTBYTES);
+
+Blake2bPersonalBytes GetHPersonalizer(uint8_t i) {
+  return {85, 65, 95, 70, 52, 74, 117, 109, 98, 108, 101, 95, 72, i, 0, 0};
+}
+
+Blake2bPersonalBytes GetGPersonalizer(uint8_t i, uint16_t j) {
+  return {85,
+          65,
+          95,
+          70,
+          52,
+          74,
+          117,
+          109,
+          98,
+          108,
+          101,
+          95,
+          71,
+          i,
+          static_cast<uint8_t>(j & 0xFF),
+          static_cast<uint8_t>(j >> 8)};
+}
+
+void FillBlake2bParamPersonal(const Blake2bPersonalBytes& personalizer,
+                              blake2b_param& params) {
+  CHECK_EQ(personalizer.size(), sizeof(params.personal));
+  memcpy(params.personal, personalizer.data(), sizeof(params.personal));
```

Comment:
I'm not seeing anyway this is going to be abused because digest_len is constrained in the params and the personalizer values are hardcoded above this. However, if someone else is able to think of a safer method to do this, I think it would still be useful.